### DISCUSSION TITLE
Is there a way for macro command to add in the '|' operator as well inside the macro itself?

### DISCUSSION THREAD

**Discussion by Sripadvallabh** [python-cmd2/cmd2#1301] [Q&A]
I just have some thoughts to create a macro along with the pipe operator like this:
macro create intdown nx | awk '$5=="down" || $6=="down"{print $0}'
but the pipe operator is taken as a pipe to the macro command itself i guess.

Is there a way to do it? i just want to store the awk inside the macro as well

**Comment by kmvanbrunt** [comment]
Yes, when you define the macro, quote pipes and redirection characters that are part of the definition.
You can get pretty detailed help by typing `help macro create`.

Also, see the docs.
https://cmd2.readthedocs.io/en/stable/features/shortcuts_aliases_macros.html#macros

**Comment by Sripadvallabh** [comment]
Thanks a lot kevin. Appreciate your quick responses. I have been more than pleased to work with you.

**Reply by Sripadvallabh** [reply]
Acutally i was going much through the documentation PDF rather than the help commands. And it seems to be an older version i guess. Thanks.
